### ISSUE TITLE
Komodo hangs when recovering mutliple windows

### ISSUE THREAD

**Issue by cgchoffman** [Komodo/KomodoEdit#171]
Logs:
https://gist.github.com/cgchoffman/f08ffac70ea2374ebe48
It seems that line [93 in codeintel.js](https://github.com/Komodo/KomodoEdit/blob/master/src/chrome/komodo/content/codeintel/codeintel.p.js#L93) is failing to ever be "ready" so codeintel is never set as "active" which leaves codeintel.js in an endless loop of trying to enable codeintel.
The "ready" state for codeintel gets set in _sent_init_requests, in it's fixup_db callback function, which in turn eventually calls initialization_complete().

From there it looks like the [get_lang_info callback](https://github.com/Komodo/KomodoEdit/blob/master/src/codeintel/src/komodo/koCodeIntel.py#L896) in _send_init_requests might be failing to retrieve particular languages information which hits a code path that would result in an endless loop of retrieving a failed languages information.  I"ve tried adding outstanding_cpln_langs.discard(lang) to line 899 to dropped the failing lang but that has no effect and I suspect  update() already does that some how.  
Several users are hitting this now.  We need a fix for 9.0.1.  Assigning to @toddw-as.

**Comment by cgchoffman** [comment]
Didn't make it clear. My attempted fix did not work.  Komodo still hangs.

**Comment by toddw-as** [comment]
Problem: There was a race condition in the codeintel manager - such that *activate* was called multiple times (possible in multiple windows startup). The *activate* method was not designed to handle this situation, so it could start re-initializing the codeintel system twice (or more), resulting in broken behaviour.

To fix this, I tweaked (protected) the *activate* method to return if it was already activated.

Note: I saw someone posting a patch that tweaked *codeintel.p.js* to set *_isActive = true* earlier, which partly solves the problem, but that still means *activate* gets called twice - and thus it can still break Komodo.
